src/mlpack/tests/activation_functions_test.cpp

```diff
@@ -1063,3 +1113,29 @@ TEST_CASE("GaussianFunctionTest", "[ActivationFunctionsTest]")
   CheckDerivativeCorrect<GaussianFunction>(desiredActivations,
                                            desiredDerivatives);
 }
+
+/**
+ * Basic test of the Softmin function.
+ */
+TEST_CASE("SoftminFunctionTest", "[ActivationFunctionsTest]")
+{
+
+  // Calculated manually using Python-numpy.
+  const arma::colvec desiredActivations("2.25051482e-43 1.24151100e-45 \
+                                         3.38351218e-46 1.00000000e+00 \
+                                         1.12046535e-44 8.27918134e-44 \
+                                         4.12196167e-45 3.04574061e-44");
+
+  /* 
+  Commented out because derivatives haven't been figured out.
+  const arma::colvec desiredDerivatives("1 1 1 1 1 1 1 0");
+  */
+
+  CheckSoftminActivationCorrect(activationData,
+                                desiredActivations);
+  /* 
+  Commented out because derivatives haven't been figured out.
+  CheckHardShrinkDerivativeCorrect(desiredActivations,
+                                   desiredDerivatives);
+  */
```

Comment:
Could you add tests for the backward as well.

Reply:
Done.

Reply:
> Could you add tests for the backward as well.

    Can you take a look now, I am still unsure about the Backward and it's tests.

Reply:
Sure, I'll take a look tonight.